Graphics/GraphicsEngineOpenGL/src/GLContextAndroid.cpp

```diff
@@ -233,14 +253,16 @@ bool GLContext::Init(ANativeWindow* window)
     //Initialize EGL
     //
     window_ = window;
-    if (window != nullptr)
+    InitEGLDisplay();
+    if (eglGetCurrentContext() != EGL_NO_CONTEXT)
     {
-        InitEGLSurface();
-        InitEGLContext();
+        AttachToCurrentEGLSurface();
+        AttachToCurrentEGLContext();
```

Comment:
Does it make sense to check the surface and context independently?
E.g., maybe there is context but no surface?

Reply:
I thought about that. Do you think AttachToCurrentEGL surface should be allowed to fail? And then try to create it?